python/GafferArnold/ArnoldTextureBake.py

```diff
@@ -0,0 +1,495 @@
+				"""
+				# Locate the next point in the list of files to bake where we can split the list into chunks without
+				# seperating two files that need to get combined into the same texture
+				def nextChunkBreak( i, l ):
+					while i > 0 and i < len( l ) and (
+							l[i - 1].get("udim") == l[i].get("udim") and
+							l[i - 1].get("fileName") == l[i].get("fileName") ):
+						i += 1
+					return i
+					
+				rawInfo = parent["__udimQuery"]["udimInformation"]
+
+				defaultFileName = parent["defaultFileName"]
+				defaultResolution = parent["defaultResolution"]
+
+				listInfo = []
+				for udim, meshes in rawInfo.items():
+					for mesh, extraAttributes in meshes.items():
+						resolution = defaultResolution
+						if "bake:resolution" in extraAttributes:
+							resolution = extraAttributes["bake:resolution"].value
+
+						fileName = defaultFileName
+						if "bake:fileName" in extraAttributes:
+							fileName = extraAttributes["bake:fileName"].value
+						
+						listInfo.append( { "udim" : int( udim ), "mesh" : mesh, "resolution" : resolution, "fileName" : fileName } )
+
+				listInfo.sort( key = lambda i: (i["fileName"], i["udim"] ) )
+
+				info = IECore.CompoundObject()
+
+				numTasks = parent["numTasks"]
+				taskIndex = parent["taskIndex"]
+
+				chunkStart = nextChunkBreak( ( taskIndex * len( listInfo ) ) / numTasks, listInfo )
+				chunkEnd = nextChunkBreak( ( ( taskIndex + 1 ) * len( listInfo ) ) / numTasks, listInfo )
+
+				dupeCount = 0
+				prevFileName = ""
+				for i in listInfo[chunkStart:chunkEnd]:
+					o = IECore.CompoundObject()
+					o["mesh"] = IECore.StringData( i["mesh"] )
+					o["udim"] = IECore.IntData( i["udim"] )
+					o["resolution"] = IECore.IntData( i["resolution"] )
+
+					udimStr = str( i["udim"] )
+					fileName = i["fileName"].replace( "<UDIM>", udimStr )
+
+					if fileName == prevFileName:
+						dupeCount += 1
+						fileName = fileName + ".layer" + str( dupeCount )
+					else:
+						prevFileName = fileName
+						dupeCount = 0
+
+					o["fileName"] = IECore.StringData( fileName )
+
+					name = o["mesh"].value.replace( "/", "_" ) + "." + udimStr
+					info[ name ] = o
+				parent["__chunkedBakeInfo"] = info
+
+				fileList = []
+				for name, i in info.items():
+					fileName = i["fileName"].value
+					for nameAndAov in parent["aovs"].strip( " " ).split( " " ):
+						fileList.append( i["fileName"].value.replace( "<AOV>", nameAndAov.split(":")[0] ) )
+				parent["renderFileList"] = IECore.StringVectorData( fileList )
+				"""
+			), "python" )
+
+
+
+			self["__parent"] = GafferScene.Parent()
+			self["__parent"]["parent"].setValue( "/" )
+			for c in ['bound', 'transform', 'attributes', 'object', 'childNames', 'setNames', 'set']:
+				self["__parent"]["in"][c].setInput( self["in"][c] )
+
+			self["__outputExpression"] = Gaffer.Expression()
+			self["__outputExpression"].setExpression( inspect.cleandoc(
+				"""
+				import IECoreScene
```

Comment:
In addition to adding the outputs we want, we should probably remove all pre-existing outputs here as well, since we don't want to render them.